File: Mcs-cs/Mcs/Models/MinimumCommonSupergraph.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Mcs.Models
{
    public class MinimumCommonSupergraph
    {
        public readonly MaximumCommonSubgraph Mcs;
        /// <summary>
        /// List of vertex mappings to minimum common supergraph. For every pair, V1 is
        /// form G1 and V2 is from G2.
        /// </summary>
        public readonly List<Pair> M;
        public readonly List<List<int>> AdjacencyMatrix;

        public MinimumCommonSupergraph(MaximumCommonSubgraph mcs, List<Pair> m, List<List<int>> adjacencyMatrix)
        {
            Mcs = mcs;
            M = m;
            AdjacencyMatrix = adjacencyMatrix;
        }
    }
}

File: mcs-java/src/main/java/anc/models/MinimumCommonSupergraph.java
package anc.models;

import java.util.List;

import com.google.common.base.Objects;

public class MinimumCommonSupergraph {
    public final MaximumCommonSubgraph mcs;
    /**
     * List of vertex mappings to minimum common supergraph. For every pair, v1 is
     * form g1 and v2 is from g2.
     */
    public final List<Pair> M;
    public final int[][] adjacencyMatrix;

    public MinimumCommonSupergraph(MaximumCommonSubgraph mcs, List<Pair> M, int[][] adjacencyMatrix) {
        this.mcs = mcs;
        this.M = M;
        this.adjacencyMatrix = adjacencyMatrix;
    }
    
    public MinimumCommonSupergraph(List<Pair> M, int[][] adjacencyMatrix) {
        this(null, M, adjacencyMatrix);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(mcs, M, adjacencyMatrix);
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        MinimumCommonSupergraph other = (MinimumCommonSupergraph) obj;
        return Objects.equal(mcs, other.mcs) && Objects.equal(M, other.M)
                && Objects.equal(adjacencyMatrix, other.adjacencyMatrix);
    }
}
